src/Simple.OData.Client.V4.Adapter/Extensions/ExtendedBoundClient.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+
+namespace Simple.OData.Client.V4.Adapter.Extensions
+{
+    /// <summary>
+    /// Provides access to extended OData operations e.g. data aggregation extensions in a fluent style.
+    /// </summary>
+    /// <typeparam name="T">The entry type.</typeparam>
+    public class ExtendedBoundClient<T> : BoundClient<T>, IExtendedBoundClient<T> where T : class
+    {
+        private ExtendedBoundClient(Session session, FluentCommand command) 
+            : base(new ODataClient(session.Settings), session, null, command)
+        {
+        }
+
+        internal ExtendedBoundClient(ODataClient oDataClient, Session session, bool dynamicResults = false)
+            : base(oDataClient, session, dynamicResults: dynamicResults)
+        {
+        }
+
+        public IExtendedBoundClient<TR> Apply<TR>(Func<IDataAggregation<T>, IDataAggregation<TR>> dataAggregation) where TR : class
+        {
+            var dataAggregationBuilder = new DataAggregationBuilder<T>(Session);
+            dataAggregation(dataAggregationBuilder);
+            AppendDataAggregationBuilder(dataAggregationBuilder);
+            if (typeof(T) != typeof(TR))
+            {
+                Session.Settings.IgnoreUnmappedProperties = true;
```

Comment:
This seems out of place as now the entire session is ignoring unmapped properties.

Reply:
I have compeletely removed changing of session settings and added a remark to the xml docs to instruct users to set this setting if necessary